src/view/TrackView.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+
+from typing import List, Optional, Type, Dict, Any
+
+import pygame
+from planar import Vec2
+from pygame.event import EventType
+from pygame.rect import Rect
+from pygame.surface import Surface
+
+from model.Simulation import Simulation
+from model.geom.track import Track
+from view import Colors
+from view.Action import Action, ActionType
+from view.View import View
+
+
+class TrackView(View):
+    simulation: Simulation
+    simulation_class: Type[Simulation]
+    dataset: Dict[str, Any]
+    board: Surface
+    coord_start: Vec2
+    scale = 1.0
+    poi = Vec2(0, 0)
+    # margin = 0
+    background_color = Colors.GRAY
+    foreground_color = Colors.BLACK
+    car_color = Colors.RED
+
+    def __init__(self, simulation: Optional[Simulation]):
+        super().__init__()
+        # to make non optional, as it really is, builder would be required
+        self.simulation = simulation
+
+    @classmethod
+    def from_dataset(
+        cls, simulation_class: Type[Simulation], dataset: Dict[str, Any]
+    ) -> TrackView:
+        tv = cls(None)
```

Comment:
It may be a little bit confusing why None is given as a 'simulation' argument. Maybe some docs in init (why is it Optional) could explain it.